### PR TITLE
Private datacenter capabilities

### PR DESCRIPTION
**Pull request by Alexander-Kita**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

pi_datacenters (d)

```
=== RUN   TestAccIBMPIDatacenterDataSourcePrivate
--- PASS: TestAccIBMPIDatacenterDataSourcePrivate (22.04s)
PASS
```

```
=== RUN   TestAccIBMPIDatacenterDataSourceBasic
--- PASS: TestAccIBMPIDatacenterDataSourceBasic (10.14s)
PASS
```

pi_datacenters (d)

```
=== RUN   TestAccIBMPIDatacentersDataSourceBasic
--- PASS: TestAccIBMPIDatacentersDataSourceBasic (13.22s)
PASS
```

```
=== RUN   TestAccIBMPIDatacentersDataSourcePrivate
--- PASS: TestAccIBMPIDatacentersDataSourcePrivate (20.12s)
PASS
```

### CODE REVIEW

File: ibm/service/power/data_source_ibm_pi_datacenter.go [hunk 1]
```diff
@@ -180,7 +185,18 @@ func dataSourceIBMPIDatacenterRead(ctx context.Context, d *schema.ResourceData,
...
+	if !sess.IsOnPrem() {
+		client = instance.NewIBMPIDatacenterClient(ctx, sess, "")
+	} else {
+		if cloudInstanceId, ok := d.GetOk(Arg_CloudInstanceID); ok {
+			client = instance.NewIBMPIDatacenterClient(ctx, sess, cloudInstanceId.(string))
+		} else {
+			return diag.Errorf("must provide %s if the datacenter is private/on-prem", Arg_CloudInstanceID)
+		}
+	}
+
```
Comment (michaelkad) [inline ibm/service/power/data_source_ibm_pi_datacenter.go:199]: Can you update it so the on premise check is done in the sdk and only have

```
var cloudInstance := ""
if cloudInstanceId, ok := d.GetOk(Arg_CloudInstanceID); ok {
			cloudInstance = cloudInstance
		} 
client = instance.NewIBMPIDatacenterClient(ctx, sess, cloudInstance)
```

And update the doc to say the instanceID is required if onprem

Comment (Alexander-Kita) [reply ibm/service/power/data_source_ibm_pi_datacenter.go]: Fixed above. Docs already specify it as required if private.

File: website/docs/d/pi_datacenter.html.markdown [hunk 2]
```diff
@@ -38,6 +38,7 @@ Example usage:
 
 Review the argument references that you can specify for your data source.
 
+- `pi_cloud_instance_id` - (Required if private datacenter, String) The GUID of the service instance associated with an account.
```
Comment (michaelkad) [inline website/docs/d/pi_datacenter.html.markdown:41]: That first part can be in the comments, like we did it in the instance
https://github.com/powervs-ibm/terraform-provider-ibm/blob/7b568a12e5926b8557ace2b12a794818fe9a3beb/website/docs/r/pi_instance.html.markdown?plain=1#L69C1-L69C2

File: website/docs/d/pi_datacenters.html.markdown [hunk 3]
```diff
@@ -30,6 +30,12 @@ Example usage:
     }
   ```
 
+## Argument reference
+
+Review the argument references that you can specify for your data source.
+
+- `pi_cloud_instance_id` - (Required if private datacenter, String) The GUID of the service instance associated with an account.
```
Comment (michaelkad) [inline website/docs/d/pi_datacenters.html.markdown:37]: Same as above
https://github.com/powervs-ibm/terraform-provider-ibm/blob/7b568a12e5926b8557ace2b12a794818fe9a3beb/website/docs/r/pi_instance.html.markdown?plain=1#L69C1-L69C2

File: website/docs/d/pi_datacenter.html.markdown (hunk 2 above)
Comment (Alexander-Kita) [reply website/docs/d/pi_datacenter.html.markdown]: Updated

**Review by michaelkad — APPROVED** [review]

**Review by ismirlia — APPROVED** [review]

**Comment by Alexander-Kita** [comment]
Opened in main repo: https://github.com/IBM-Cloud/terraform-provider-ibm/pull/5833

**Pull request closed without merging**